### PR TITLE
Adding GC deployment implementation

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
On every reconciliation loop, the controller updates/creates GarbageCollector deployment, based on the NFD CR spec

### CODE REVIEW

File: internal/controllers/nodefeaturediscovery_reconciler.go [hunk 1]
```diff
@@ -204,6 +204,17 @@ func (nfdh *nodeFeatureDiscoveryHelper) handleTopology(ctx context.Context, nfdI
...
 func (nfdh *nodeFeatureDiscoveryHelper) handleGC(ctx context.Context, nfdInstance *nfdv1.NodeFeatureDiscovery) error {
+	gcDep := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{Name: "nfd-gc", Namespace: nfdInstance.Namespace},
+	}
+	opRes, err := controllerutil.CreateOrPatch(ctx, nfdh.client, &gcDep, func() error {
+		return nfdh.deploymentAPI.SetGCDeploymentAsDesired(nfdInstance, &gcDep)
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to reconcile master deployment %s/%s: %w", nfdInstance.Namespace, nfdInstance.Name, err)
```
[the remark this answers is not in the document]
Comment (yevgeny-shnaidman) [reply internal/controllers/nodefeaturediscovery_reconciler.go]: Thanks, fixed

**Review by marquiz — APPROVED** [review]
Looks fine to me, now 👍

/assign @ArangoGutierrez

**Comment by yevgeny-shnaidman** [comment]
ping @ArangoGutierrez

**Pull request merged**